### ISSUE TITLE
Add a --fix option for quote-props

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#6996]
I'm using Node v6.4.0 and ESLint v3.4.0.

It would be useful to have an autofix option for the `quote-props` rule.

* If a property is reported because quotes are present and unnecessary, the fixer will remove the quotes.
* If a property is reported because quotes are missing but should be present, the fixer will add quotes. It should probably add double quotes by default (for consistency with the `quotes` rule), and then let the `quotes` rule clean up on the next pass if the user prefers another type of quotes. The fixer will need to account for cases where number literals are used as keys:

```js
var myObject = {1e2: "foo", 0x123: "bar", 123.: "baz"};

// should get fixed to:

var myObject = {"100": "foo", "291": "bar", "123": "baz"};

// but should NOT get fixed to:

var myObject = {"1e2": "foo", "0x123": "bar", "123.": "baz"}
```

* If the `consistent` option is used and errors are reported, I see 3 possible ways to fix the errors:
  1. Add quotes to all the unquoted properties. (I think this option is probably best)
  1. Remove quotes from all quoted properties if possible, otherwise add quotes to all unquoted properties. (I don't think this option is a good idea, because if people want this behavior they can just use `consistent-as-needed`)
  1. Use some heuristic (e.g. the first property) to determine whether to quote everything or to quote nothing.
  

I would be willing to add this if the issue is accepted.

**Comment by platinumazure** [comment]
I think I would prefer not to autofix numeric properties, just because it could confuse users. I'm willing to :+1: endorse either way.

**Comment by mysticatea** [comment]
:+1: 
The option i. of `consistent` sounds good to me.

Also, the current behavior of `quote-props` rule seems to report the correct name about numeric names.

![image](https://cloud.githubusercontent.com/assets/1937871/18031681/7138825c-6d27-11e6-9cc1-6f9331468d06.png)

So I think the rule should modify the numeric names to quoted names automatically. If a developer wants to not warn the numeric names, it needs an option for that.

**Comment by nzakas** [comment]
Let's just discuss directly on #7095 in order to avoid having the conversation in two places.
